tests/rest/client/test_login.py

```diff
@@ -384,6 +385,38 @@ def test_session_can_hard_logout_all_sessions_after_being_soft_logged_out(
         channel = self.make_request(b"POST", "/logout/all", access_token=access_token)
         self.assertEqual(channel.result["code"], b"200", channel.result)
 
+    def test_login_with_overly_large_device_id_fails(self) -> None:
+        self.register_user("mickey", "cheese")
+
+        # create a device_id larger than 512B
+        device_id = 1000**1000
```

Comment:
Note that `device_id` is an integer here rather than a string. I expect this will get encoded as a JSON number, which is contrary to the spec!

To make large strings, one can use the `*` operator: e.g.

```python
>>> "hello"*50
'hellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohellohello'
```